doc/dev/cephadm/host-maintenance.rst

```diff
@@ -0,0 +1,100 @@
+================
+Host Maintenance
+================
+
+All hosts that support Ceph daemons need to support maintenance activity, whether the host
+is physical or virtual(vm or cloud). This means that management workflows should provide
+a simple and consistent way to support this operational requirement. This document defines
+the maintenance strategy that could be implemented in cephadm and mgr/cephadm.
+
+
+High Level Design
+=================
+Placing a host into maintenance, adopts the following workflow;
+
+#. confirm that the removal of the host does not impact data availability (the following
+   steps will assume it is safe to proceed)
+
+   * orch host ok-to-stop <host> would be used here
```

Comment:
Please make this operation more accurate than `osd` `ok-to-stop` and `safe-to-destroy` have been. Those tools have been overly paranoid in ways that have limited their utility.

Reply:
ok-to-stop is what we have currently - have you opened issues against ok-top-stop? If that's not adequate we should work on fixing those tools and not working around them.